tests/components/tplink/test_init.py

```diff
@@ -209,13 +232,141 @@ async def test_platforms_are_initialized(hass):
     ) as switch_setup, patch(
         "homeassistant.components.tplink.common.SmartPlug.is_dimmable", False
     ):
+
+        light = SmartBulb("123.123.123.123")
+        switch = SmartPlug("321.321.321.321")
+        switch.get_sysinfo = MagicMock(return_value=SMARTPLUGSWITCH_DATA["sysinfo"])
+        switch.get_emeter_realtime = MagicMock(
+            return_value=EmeterStatus(SMARTPLUGSWITCH_DATA["realtime"])
+        )
+        switch.get_emeter_daily = MagicMock(
+            return_value={int(time.strftime("%e")): 1.123}
+        )
+        get_static_devices.return_value = SmartDevices([light], [switch])
+
         # patching is_dimmable is necessray to avoid misdetection as light.
         await async_setup_component(hass, tplink.DOMAIN, config)
         await hass.async_block_till_done()
 
-    assert discover.call_count == 0
-    assert light_setup.call_count == 1
-    assert switch_setup.call_count == 1
+        assert hass.data.get(tplink.DOMAIN)
```

Comment:
Please don't assert integration details like hass.data in the tests. Assert core state.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations